orderer/common/multichannel/registrar.go

```diff
@@ -430,8 +450,42 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block) (types.
 		return types.ChannelInfo{}, types.ErrChannelAlreadyExists
 	}
 
-	//TODO
-	return types.ChannelInfo{}, errors.New("Not implemented yet")
+	if !isAppChannel && len(r.chains) > 0 {
+		return types.ChannelInfo{}, types.ErrAppChannelsAlreadyExists
+	}
+
+	configEnv, err := protoutil.ExtractEnvelope(configBlock, 0)
+	if err != nil {
+		return types.ChannelInfo{}, errors.Wrap(err, "failed extracting config envelope from block")
+	}
+
+	//TODO save the join-block in the file repo to make this action crash tolerant.
```

Comment:
do we have an API that defines how to do that? If yes can we put this here?

Reply:
We are working on that in parallel, it is defined in tasks:
https://jira.hyperledger.org/browse/FAB-17960,
https://jira.hyperledger.org/browse/FAB-18009,
https://jira.hyperledger.org/browse/FAB-18008.

After a discussion with the ledger team & Jason Yellick, the plan is to use an "atomic file repo" to save the join blocks. The repo will be scanned on startup to reconcile the join state of chains with their ledger. 

BTW, the same file repo code will be used to make ledger removal CFT, see here:
https://jira.hyperledger.org/browse/FAB-17868